pkg/migration-tool/utils/client.go

```diff
@@ -119,26 +195,47 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query, desc string) (re
 	}
 	compressed, err = ioutil.ReadAll(bytes.NewReader(reader.Bytes()))
 	if err != nil {
-		return nil, -1, -1, errors.Wrap(err, fmt.Sprintf("error reading response. HTTP status code: %s", httpResp.Status))
+		err = errors.Wrap(err, fmt.Sprintf("error reading response. HTTP status code: %s", httpResp.Status))
+		if shouldRetry(c.cRuntime.OnErr, c.cRuntime, numAttempts, fmt.Sprintf("error:\n%s\n", err.Error())) {
```

Comment:
couldn't the error here be a timeout, shouldn't we check for that?

Reply:
I don't think so. The errors here would be related to reading the underlying byte slice.